### PR TITLE
First version of RoundedButton

### PR DESCRIPTION
**Pull request by f2rf2r**

### CODE REVIEW

File: roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            int defaultFillColorRes =
+                    strokeColor == INVALID_COLOR ? R.color.rb_default_fill_color : android.R.color.transparent;
+            int defaultFillColor = ContextCompat.getColor(context, defaultFillColorRes);
+            fillColor = typedArray.getColor(R.styleable.RoundedButton_rb_fill_color, defaultFillColor);
+
+            disabledColor = typedArray.getColor(R.styleable.RoundedButton_rb_disabled_color, ContextCompat.getColor(getContext(), R.color.rb_default_disabled_color));
+
+            typedArray.recycle();
+        }
+        setGravity(Gravity.CENTER);
```
Comment (deanshi) [inline roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java]: Does this mean we can't set the gravity of our text inside the button?

Comment (deanshi) [reply roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java]: I mean I can't think of a time I wouldn't want it to be centered, but is it noted anywhere?

File: roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        updateBackground();
+    }
+
+    private void updateBackground() {
+        StateListDrawable stateListDrawable = new StateListDrawable();
+        Drawable defaultDrawable = createRoundedRectangleDrawable(fillColor, strokeColor, strokeWidth);
+        int darkenedFill = darkenColor(fillColor);
+        Drawable pressedDrawable = createRoundedRectangleDrawable(darkenedFill, strokeColor, strokeWidth);
+        Drawable disabledDrawable = createRoundedRectangleDrawable(disabledColor, strokeColor, strokeWidth);
+        stateListDrawable.addState(new int[] { -android.R.attr.state_enabled }, disabledDrawable);
```
Comment (deanshi) [inline roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java]: I haven't looked too deep, but is there really no other way of handling the disabled state? That seems kind of weird.

Comment (f2rf2r) [reply roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java]: as in the negative sign? that's the standard way of setting the false state

File: roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java (hunk 1 above)
Comment (f2rf2r) [reply roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java]: I updated it to allow it to be changed while still defaulting to center if it's not set

File: roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java (hunk 2 above)
Comment (deanshi) [reply roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java]: Huh, I've never seen that before. Okay good to know.

File: roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            rectDrawable.setStroke(strokeWidth, strokeColor);
+        }
+        rectDrawable.setColor(fillColor);
+        // using some really large radius here, Android will automatically clip it down to height/2
+        rectDrawable.setCornerRadius(10000);
+        return rectDrawable;
+    }
+
+    // Create a slightly darkened color for touch indicator
+    private int darkenColor(int originalColor) {
```
Comment (deanshi) [inline roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java:126]: If the user has a black/darker background, would this still show anything?

**Comment by deanshi** [comment]
Other then those comments, and potential gradle things, everything looks good to me :shipit:

File: roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java (hunk 3 above)
Comment (f2rf2r) [reply roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java:126]: darker fills would have a less visible pressed state. I just added an option to manually set the pressed state color

**Review by deanshi — APPROVED** [review]

**Pull request merged**